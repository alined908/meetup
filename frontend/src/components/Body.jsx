import React, {Component} from 'react';
import {Route, Switch} from 'react-router-dom';
import LogoutComponent from "./auth/LogoutComponent";
import ErrorComponent from "./auth/ErrorComponent";
import RegisterComponent from "./auth/RegisterComponent";
import WelcomeComponent from "./WelcomeComponent";
import ChatComponent from "./chat/ChatComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import HomeComponent from "./HomeComponent";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import FriendsComponent from './invite/FriendsComponent';
import MeetupEventForm from "./meetup/MeetupEventForm"
import AuthenticatedRoute from "../accounts/AuthenticatedRoute";
import LoginComponent from "./auth/LoginComponent";
import Invites from "./invite/Invites"
import MeetupForm from "./meetup/MeetupForm"
import Profile from "./Profile"

class Body extends Component {

    render () {
        return (
            <div className="c">
                <Switch>
                    <Route path="/" exact component={HomeComponent}></Route>
                    <Route path="/login" component={LoginComponent}></Route>
                    <Route path="/register" component={RegisterComponent}></Route>
                    <AuthenticatedRoute path="/logout" component={LogoutComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/welcome" component={WelcomeComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri/new" exact component={MeetupEventForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/new" exact component={MeetupForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri" component={MeetupPageComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups" exact component={MeetupsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat/:uri" component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat" exact component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/friends" component={FriendsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/invites" component={Invites}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/profile/:id" component={Profile}></AuthenticatedRoute>
                    <Route component={ErrorComponent}></Route>
                </Switch>
            </div>
        )
    }
}

export default Body