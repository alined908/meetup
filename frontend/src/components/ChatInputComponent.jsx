import React, {Component} from 'react'
import {connect} from 'react-redux';
import {setTypingValue, addMessage} from "../actions/chat";


class ChatInputComponent extends Component{
    handleChange = (e) => {
        console.log(e.target.value);
        this.props.setTypingValue(e.target.value);        
    }

    handleSubmit = (e) => {
        if (e.key === "Enter"){
            e.preventDefault();
            this.props.addMessage(e.target.value, this.props.room)
        }
    }

    render(){
        return (
            <form className="chat-input-form">
                <input className="chat-input"
                    type="text"
                    onChange={this.handleChange} 
                    onKeyPress={this.handleSubmit}
                    value={this.props.message} 
                    placeholder="Type here to send a message">
                </input>
            </form>
        )
    } 
}

function mapStateToProps(state){
    return {
        message: state.chat.setTypingValue,
        room: state.chat.activeRoom
    }
}

const mapDispatchToProps = {
    setTypingValue,
    addMessage
}

export default connect(mapStateToProps, mapDispatchToProps)(ChatInputComponent)