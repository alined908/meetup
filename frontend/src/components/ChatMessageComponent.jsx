import React, {Component} from 'react'
import moment from 'moment';

class ChatMessageComponent extends Component {
    render (){
        const [id, message, timestamp, is_read, room, sender, receipient] = Object.values(this.props.message)
        console.log(this.props.user)
        let user = JSON.parse(this.props.user);

        var is_user_msg = user.id === sender
    
        if (is_user_msg){
            var name = user.first_name
        } else {
            var name = this.props.members[sender].first_name
        }

        return (
            <div className={`chat-msg-wrapper ${is_user_msg ? "is-user": ""}`}>
                <div className={`chat-msg-info ${is_user_msg ? "is-user": ""}`}>
                    <div className="chat-msg-id">
                        {name}
                    </div>
                    <div className="chat-msg-time">
                        {moment(timestamp).local().format("MMM DD hh:mm A")}
                    </div>
                </div>
                <div className={`chat-msg ${is_user_msg ? "is-user": ""}`}>{message}</div>
            </div>
        )
    }
}


export default ChatMessageComponent