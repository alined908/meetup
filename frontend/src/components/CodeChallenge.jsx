import React, {Component} from 'react'
import {axiosClient} from '../accounts/axiosClient'
import styles from "../styles/challenge.module.css"
import {DragDropContext, Droppable, Draggable} from 'react-beautiful-dnd'
import {CircularProgress} from '@material-ui/core'
import CloseIcon from '@material-ui/icons/Close';

const copy = (filled, destination, numCategories) => {
    const col = Number(destination.droppableId.split("-")[1]) % numCategories
    return Object.assign([], filled, {[col]: true})
}

const move = (filled, source, destination, numCategories) => {
    const oldCol = Number(source.droppableId.split("-")[1]) % numCategories
    const newCol = Number(destination.droppableId.split("-")[1]) % numCategories
    return Object.assign([], filled, {[oldCol]: false, [newCol]: true})
}   

const getStyle = (style, snapshot) => {
    if (!snapshot.isDragging) return {};
    if (!snapshot.isDropAnimating) {
      return style;
    }
    return {
      ...style,
      transitionDuration: `0.001s`,
    };
  }
  
class Card extends Component {
    constructor(props){
        super(props);
        this.state = {
            isHovered: false
        }
    }

    render () {
        const restaurant = this.props.restaurant
        return (
            <div className={`${styles.rstCard} ${this.state.isHovered ? styles.closeHovered : ""}`}>
                {!this.props.isMenu && 
                    <CloseIcon
                        size="small" 
                        color="secondary"
                        className={styles.closeIcon}
                        onClick={this.props.handleDelete} 
                        onMouseEnter={() => this.setState({isHovered: true})} 
                        onMouseLeave={() => this.setState({isHovered: false})}
                    />
                }
                <div className={styles.rstCardTop}>
                    <img className={styles.rstImg} src={restaurant.yelp_image} />
                </div>
                <div className={styles.rstCardBottom}>
                    <span>{restaurant.name}</span>
                    <div className={styles.rstCardCategories}>
                        {this.props.isMenu && this.props.categories.map((category, index) => 
                            this.props.filled[index] && <img
                                key={category.id} 
                                className={styles.categoryPicture} 
                                src={`${process.env.REACT_APP_S3_STATIC_URL}${category.api_label}.png`}
                            />
                        )}
                    </div>
                </div>
            </div>
        )
    }
}

class Restaurant extends Component {

    render () {
        const restaurant = this.props.restaurant

        return (
            <Draggable draggableId={this.props.isMenu ? `restaurant-${restaurant.id}`: `category-${this.props.index}`} index={this.props.index}>
                {(provided, snapshot) => (
                    <React.Fragment>
                        <div 
                            ref={provided.innerRef} 
                            {...provided.draggableProps}
                            {...provided.dragHandleProps}
                            className={styles.restaurant}
                            style={getStyle(provided.draggableProps.style, snapshot)}
                        >   
                            <Card 
                                restaurant={restaurant} 
                                isMenu={this.props.isMenu}
                                handleDelete={this.props.handleDelete}
                                filled={this.props.filled}
                                categories={this.props.categories}
                            />
                        </div>
                        {snapshot.isDragging && this.props.isMenu && (
                            <div className={styles.restaurant}>
                            
                            </div>
                        )}
                    </React.Fragment>
                )}
                
            </Draggable>
        )
    }
}

class RestaurantRow extends Component {
    
    constructor(props){
        super(props);
        this.state = {
            filled: new Array(props.categories.length).fill(false)
        }
    }

    handleDelete = (index) => {
        const newFilled = Object.assign([], this.state.filled, {[index]: false})
        this.setState({filled: newFilled})
    }

    onDragEnd = result => {
        const {source, destination} = result;

        if (!destination){
            return;
        }  

        const numCategories = this.props.categories.length;
        let filled;

        if (source.droppableId.includes("restaurant")) {
            filled = copy(this.state.filled, destination, numCategories)
        } else if (source.droppableId.includes("category")){
            filled = move(this.state.filled, source, destination, numCategories)
        }
        
        this.setState({filled})
    }

    render () {
        const restaurant = this.props.restaurant
        const categories = this.props.categories
        const offset = restaurant.id * categories.length

        return (
            <DragDropContext onDragEnd={this.onDragEnd}>
                <div className={styles.restaurantRow}>
                    <Droppable droppableId={`restaurant-${restaurant.id}`} isDropDisabled={true}>
                        {(provided, snapshot) => (
                            <div ref={provided.innerRef} {...provided.droppableProps} className={styles.restaurantWrapper}>
                                <Restaurant 
                                    isMenu={true}
                                    filled={this.state.filled}
                                    restaurant={restaurant}
                                    categories={categories}
                                    index={this.props.rstIndex}
                                />
                                <span style={{display: "none"}}>
                                    {provided.placeholder}
                                </span>
                            </div>
                        )}
                    </Droppable>
                    <div className={styles.columns}>
                        {categories.map((category, index) => 
                            <Droppable key={category.id} droppableId={`category-${offset + index}`}>
                                {(provided, snapshot) => (
                                    <div ref={provided.innerRef} className={styles.block}>
                                        {this.state.filled[index] ?  
                                            <Restaurant 
                                                isMenu={false}
                                                restaurant={restaurant}
                                                handleDelete={() => this.handleDelete(index)}
                                                index={offset + index} />
                                            :   
                                            <div className={styles.category}/>
                                        }
                                        <span style={{display: "none"}}>
                                            {provided.placeholder}
                                        </span>
                                    </div>  
                                )}
                            </Droppable>
                        )}
                    </div>
                </div>
            </DragDropContext>
        )
    }
}

class CodeChallenge extends Component {

    constructor(){
        super();
        this.state = {
            restaurants: [],
            categories: [],
            isFetching: true,
            isInitialized: false
        }
    }

    async componentDidMount() {

        const [restaurantResponse, categoryResponse] = await Promise.all([
            axiosClient.get(`/api/restaurants/challenge/`),
            axiosClient.get(`/api/categories/challenge/`)
        ])
        this.setState({
            restaurants: restaurantResponse.data, 
            categories: categoryResponse.data,
            isFetching: false,
            isInitialized: true
        })
    }

    render () {
        return (
            <>
                {this.state.isInitialized ?
                    <div className={`${styles.outerWrapper}`}>
                        <div className={styles.topWrapper}>
                            <div className={styles.restaurantsWrapper}>
                                <div className={styles.header}>
                                    Restaurants
                                </div>
                                <div className={styles.restaurantIndent}/>
                            </div>
                            <div className={styles.categoriesWrapper}>
                                <div className={styles.header}>
                                    Categories
                                </div>
                                <div className={styles.categories}>
                                    {this.state.categories.map((category, index) => 
                                        <div key={index} className={styles.categoryHeaderWrapper}>
                                            <div className={styles.categoryHeader}>
                                                <img 
                                                    className={styles.categoryPicture} 
                                                    src={`${process.env.REACT_APP_S3_STATIC_URL}${category.api_label}.png`}
                                                />
                                                {category.label}
                                            </div>
                                        </div>
                                    )}
                                </div>
                            </div>
                        </div>
                        <div className={styles.bottomWrapper}>
                            <div className={styles.restaurants}>
                                {this.state.restaurants.map((restaurant, index) =>
                                    <RestaurantRow 
                                        key={restaurant.id}
                                        rstIndex={index}
                                        restaurant={restaurant} 
                                        categories={this.state.categories}
                                    />
                                )}
                            </div>
                        </div>
                    </div>
                    :
                    <div className="loading">
                        <CircularProgress size={30}/>
                    </div>
                    
                }       
            </>
        )
    }
}

export default CodeChallenge