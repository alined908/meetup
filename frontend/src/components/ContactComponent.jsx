import React, {Component} from 'react'
import {connect} from 'react-redux';
import {setActiveRoom} from '../actions/chat';

class ContactComponent extends Component {
    handleClick = (uri) => {
        console.log("handle click")
        this.props.setActiveRoom(uri);
    }
    
    render (){
        const [id, uri, name, timestamp, members] = this.props.room

        return (
            <div className="chat-contact" onClick={() => this.handleClick(uri)}>
                <div>{name}</div>
                <div>{Object.keys(members).length + " Members"}</div>
            </div>
        )
    }
}
const mapDispatchToProps = {
    setActiveRoom
}

export default connect(null, mapDispatchToProps)(ContactComponent)