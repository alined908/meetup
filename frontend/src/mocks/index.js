import moment from 'moment'

//Category
export const category = {
    id: 1, 
    image: null,
    label: "Dessert", 
    api_label: "desserts"
}

//User
export const user = {
    id: 1, 
    email: "daniel@gmail.com", 
    first_name: "Daniel", 
    last_name: "Lee",
    avatar: null
}
export const user2 = {
    id: 2, 
    email: "example2@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const user3 = {
    id: 3, 
    email: "example3@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const settings = {
    radius: 25, 
    location: null, 
    latitude: null,
    longitude: null
}

export const userWithSettings = {
    ...user,
    settings: {...settings}
}
export const friend = {
    id: 1, 
    user: user2, 
    created_at: moment().format(), 
    chat_room: "something"
}
export const friends = [friend]
export const preference = {
    id: 1, 
    user, 
    category, 
    name: "Dessert", 
    ranking: 1, 
    timestamp: moment().format()
}
export const preferences = [preference]
export const friendInvite = {
    id: 1, 
    timestamp: moment().format(), 
    status: 1, 
    uri: "random", 
    sender: user, 
    receiver: user2
}

//Chat
export const message = {
    id: 1, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: false
}
export const message2 = {
    id: 2, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: true
}
export const messages = [message, message2]
export const members = {1: user, 2: user2}
export const room = {
    uri: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    }
}
export const rooms = {
    friend: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    },
    meetup: {
        id: 2, 
        uri: "xyz", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: null, 
        meetup:1, 
        notifs: 3
    }
}

//General
export const error = {
    message: "Something went wrong."
}

//Meetup
export const meetupMember1 = {
    1: {
        id: 1,
        user,
        ban: false,
        admin: true
    }
}

export const meetupMember2 = {
    2: {
        id: 2,
        user: user2,
        ban: false,
        admin: false
    }
}

export const meetupMember3 = {
    3: {
        id: 3,
        user: user3,
        ban: false,
        admin: false
    }
}

export const meetupMembers = {...meetupMember1, ...meetupMember2}

export const meetup = {
    uri: {
        id: 1, 
        name: "Meetup", 
        uri: "uri", 
        creator: user, 
        location: "location", 
        date: moment().format("YYYY-MM-DD"),
        members: meetupMembers,
        notifs: 0,
        public: true,
        categories: [category],
        latitude: 34.228754, 
        longitude: -118.2351192,
        events: {},
        isMeetupEventsInitialized : true
    }
}

export const meetups = {...meetup}

export const event = {
    1: {
        id: 1,
        meetup: 1,
        creator: user,
        title: "Dessert",
        start: moment().format(),
        end: moment().format(),
        chosen: null,
        categories: [category],
        options: {},
        price: "1, 2",
        distance: 16000,
        entries: {},
        random: true
    }
}
export const events = {...event}

export const option = {
    1 : {
        id: 1,
        score: 0,
        option: null,
        restaurant: {},
        votes: {},
        banned: false
    }
}

export const options = {...option}

export const tokens = {
    access: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTg4MjI5OTEzLCJqdGkiOiIwOTg2YTg0MDA1ZGM0OGQ1ODEzYmIyNzE1OWViNGUyZCIsInVzZXJfaWQiOjEsInVzZXIiOnsiaWQiOjEsImVtYWlsIjoiZGFuaWVsQGdtYWlsLmNvbSIsImZpcnN0X25hbWUiOiJEYW5pZWwiLCJsYXN0X25hbWUiOiJMZWUiLCJhdmF0YXIiOm51bGwsInNldHRpbmdzIjp7InJhZGl1cyI6MjUsImxvY2F0aW9uIjpudWxsLCJsYXRpdHVkZSI6bnVsbCwibG9uZ2l0dWRlIjpudWxsfX19.FbffRthOKX5F0TGvqA5mZXuXUWNkUojzKexDGCoP518",
    refresh : "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTU4ODM4Mjk0NiwianRpIjoiN2ZhNGY2MDE4OWM3NDA5ZmFjOTMwYmE4ZTRlMWFhMTMiLCJ1c2VyX2lkIjoxLCJ1c2VyIjp7ImlkIjoxLCJlbWFpbCI6ImRhbmllbEBnbWFpbC5jb20iLCJmaXJzdF9uYW1lIjoiRGFuaWVsIiwibGFzdF9uYW1lIjoiTGVlIiwiYXZhdGFyIjpudWxsLCJzZXR0aW5ncyI6eyJyYWRpdXMiOjI1LCJsb2NhdGlvbiI6bnVsbCwibGF0aXR1ZGUiOm51bGwsImxvbmdpdHVkZSI6bnVsbH19fQ.qqaY4Ye7rmyOKLbDS2EKz4SGWH0fwZ0MvrGP6NJ3IRE"
}