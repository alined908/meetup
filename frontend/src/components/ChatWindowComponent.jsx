import React, {Component} from 'react'
import ChatMessageComponent from "./ChatMessageComponent"
import ChatInputComponent from "./ChatInputComponent"
import {connect} from 'react-redux'

class ChatWindowComponent extends Component {

    componentDidMount() {
        console.log("mounted chat window")
        this.scrollToBottom();
      }
    componentDidUpdate() {
        console.log("updated chat window")
        this.scrollToBottom();
    }
    scrollToBottom = () => {
        console.log("scroll to bottom called")
        this.messagesEnd && this.messagesEnd.scrollIntoView({ behavior: "smooth" });
    };
    
    render () {
        return (
            <div className="chat-window" ref={this.chatsRef}>
                <div className="chat-header">
                    Conversation for uri: {this.props.activeRoom}
                </div>
                <div className="chat-messages">
                    {this.props.messages && this.props.messages.map((msg) => <ChatMessageComponent user={this.props.user} message={msg.message} members={this.props.activeChatMembers}/>)}
                </div>
                {this.props.activeChatMembers && <div ref={(el) => { this.messagesEnd = el; }} className="chat-input">
                    <ChatInputComponent/>
                </div>}
            </div>
        )
    }
}

function mapStateToProps(state){
    if (state.chat.activeRoom in state.chat.rooms){
        return {
            activeChatMembers: state.chat.rooms[state.chat.activeRoom].members,
            user: state.auth.user
        }
    } else {
        return {
            activeChatMembers: null,
            user: state.auth.user
        }
    }
    
}


export default connect(mapStateToProps)(ChatWindowComponent)