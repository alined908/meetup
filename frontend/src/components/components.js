//Auth
import ErrorComponent from "./auth/ErrorComponent";
import LoginComponent from "./auth/LoginComponent";
import LogoutComponent from "./auth/LogoutComponent";
import RegisterComponent from "./auth/RegisterComponent";

//Chat
import ChatBarComponent from "./chat/ChatBarComponent"
import ChatComponent from "./chat/ChatComponent"
import ChatMessageComponent from "./chat/ChatMessageComponent"
import ChatWindowComponent from "./chat/ChatWindowComponent"
import ContactComponent from "./chat/ContactComponent"

//invite
import Friend from './invite/Friend';
import FriendsComponent from './invite/FriendsComponent';
import Invite from "./invite/Invite"
import Invites from "./invite/Invites"

//meetup
import Map from "./meetup/Map"
import Meetup from "./meetup/Meetup";
import MeetupCard from "./meetup/MeetupCard";
import MeetupEvent from "./meetup/MeetupEvent";
import MeetupEventForm from "./meetup/MeetupEventForm"
import MeetupForm from "./meetup/MeetupForm";
import MeetupFriend from "./meetup/MeetupFriend";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import Restauraunt from "./meetup/Restauraunt";

//fields
import CategoryAutocomplete from "./fields/CategoryAutocomplete"
import Location from "./fields/Location"
import renderTextField from "./fields/renderTextField"
import renderDatePicker from "./fields/renderDatePicker"
import renderDateSimplePicker from "./fields/renderDateSimplePicker"
import UserAutocomplete from "./fields/UserAutocomplete"

//General
import Body from "./general/Body"
import GlobalMessage from "./general/GlobalMessage"
import HomeComponent from "./general/HomeComponent";
import LiveUpdatingBadge from "./general/LiveUpdatingBadge"
import Navigation from "./general/Navigation"
import Profile from "./general/Profile"
import GroupAvatars from './general/GroupAvatars'
import NotificationWrapper from "./general/NotificationWrapper"
import CalendarComponent from "./general/CalendarComponent"
import CategoriesComponent from "./general/CategoriesComponent"
import CategoryComponent from "./general/CategoryComponent"
import Preferences from "./general/Preferences"
import Preference from "./general/Preference"
import SettingsComponent from "./general/SettingsComponent"

export {
    ErrorComponent, LoginComponent, LogoutComponent, RegisterComponent, 
    ChatBarComponent, ChatComponent, ChatMessageComponent, ChatWindowComponent, ContactComponent,
    Friend, FriendsComponent, Invite, Invites,
    Map, Meetup, MeetupCard, MeetupEvent, MeetupEventForm, MeetupFriend, MeetupForm, MeetupPageComponent, MeetupsComponent, Restauraunt,
    CategoryAutocomplete, UserAutocomplete, Location, renderTextField, renderDatePicker, renderDateSimplePicker,
    Body, GlobalMessage, GroupAvatars, HomeComponent, LiveUpdatingBadge, Navigation, Profile, NotificationWrapper, 
    Preferences, Preference, CalendarComponent, CategoriesComponent, CategoryComponent, SettingsComponent}