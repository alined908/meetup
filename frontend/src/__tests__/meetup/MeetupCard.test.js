import React from "react";
import { shallow, mount } from "enzyme";
import { MeetupCard } from "../../components/components";
import * as mocks from "../../mocks";

const props = {
  meetup: mocks.meetup.uri,
};

describe("MeetupCard", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders without crashing given props", () => {
    const wrapper = shallow(<MeetupCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("shows events section with categories if meetup has categories", () => {
    const wrapper = shallow(<MeetupCard {...props} />);
    expect(wrapper.find(".cardEvents")).toHaveLength(1);
    expect(wrapper.find(".cardEventsCategory").text()).toContain("Dessert");
  });

  it("doesnt show events section if meetup has no categories", () => {
    const propsNoCategories = { meetup: mocks.meetupWithoutCategories.uri };
    const wrapper = shallow(<MeetupCard {...propsNoCategories} />);
    expect(wrapper.find(".cardEventsCategory")).toHaveLength(0);
  });

  it("shows how many members in meetup", () => {
    const wrapper = shallow(<MeetupCard {...props} />);
    const members = wrapper.find(".cardMembers");
    expect(members.text()).toBe("2 members");
  });
});
