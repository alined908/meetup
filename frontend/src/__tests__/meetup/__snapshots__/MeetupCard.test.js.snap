// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetupCard renders without crashing given props 1`] = `
<Link
  to="/meetups/uri"
>
  <div
    className="makeStyles-paper-1 elevate"
  >
    <div
      className="cardInner"
    >
      <div
        className="cardTop"
      >
        <div
          className="cardIcon flexClip2"
        >
          <WithStyles(ForwardRef(Tooltip))
            title="Meetup"
          >
            <WithStyles(ForwardRef(Typography))
              className="flexEllipse"
              variant="h5"
            >
              Meetup
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Tooltip))>
        </div>
        <div
          className="cardIcon"
        >
          <TodayIcon />
          Saturday, May 9
        </div>
      </div>
      <div
        className="makeStyles-middle-2 cardEvents"
      >
        <div
          className="cardEventsHeader"
        >
          Events
        </div>
        <div
          className="cardEventsCategories"
        >
          <div
            className="cardEventsCategory"
            key="1"
          >
            <WithStyles(ForwardRef(Avatar))
              src="undefineddesserts.png"
              style={
                Object {
                  "height": 28,
                  "width": 28,
                }
              }
              variant="square"
            />
            Dessert
          </div>
        </div>
      </div>
      <div
        className="cardBottom"
      >
        <WithStyles(ForwardRef(Tooltip))
          title="location"
        >
          <div
            className="cardIcon flexClip"
          >
            <RoomIcon />
            <div
              className="flexEllipse"
            >
              location
            </div>
          </div>
        </WithStyles(ForwardRef(Tooltip))>
        <div
          className="cardIcon flexClip"
        >
          <Connect(GroupAvatars)
            members={
              Array [
                Object {
                  "avatar": null,
                  "email": "daniel@gmail.com",
                  "first_name": "Daniel",
                  "id": 1,
                  "last_name": "Lee",
                },
                Object {
                  "avatar": null,
                  "email": "example2@gmail.com",
                  "first_name": "Bob",
                  "id": 2,
                  "last_name": "Jim",
                },
              ]
            }
          />
          <span
            className="flexEllipse cardMembers"
          >
            2 member
            s
          </span>
        </div>
      </div>
    </div>
  </div>
</Link>
`;
